shell/browser/api/electron_api_app.cc

```diff
@@ -779,18 +779,19 @@ void App::AllowCertificateError(
     bool is_main_frame_request,
     bool strict_enforcement,
     base::OnceCallback<void(content::CertificateRequestResultType)> callback) {
-  auto adapted_callback = base::AdaptCallbackForRepeating(std::move(callback));
   v8::Isolate* isolate = JavascriptEnvironment::GetIsolate();
   v8::Locker locker(isolate);
   v8::HandleScope handle_scope(isolate);
-  bool prevent_default =
-      Emit("certificate-error",
-           WebContents::FromOrCreate(isolate, web_contents), request_url,
-           net::ErrorToString(cert_error), ssl_info.cert, adapted_callback);
+  auto callback_pair = base::SplitOnceCallback(std::move(callback));
+  bool prevent_default = Emit("certificate-error",
+                              WebContents::FromOrCreate(isolate, web_contents),
+                              request_url, net::ErrorToString(cert_error),
+                              ssl_info.cert, std::move(callback_pair.first));
```

Comment:
I don't think we can pass one of a pair of split OnceCallbacks to JS, because if the JS holds onto the function and calls it later, after we've already called the twin, the process will crash. We need our own version of `AdaptCallbackForRepeating` here.

Though tbh this particular API is a bit suspect, I'm not sure we should be both passing a callback _and_ checking preventDefault? In any case, we shouldn't change the API in the roll, but this needs to not hard-crash the process no matter when/how many times this callback is called from JS.

Reply:
OK. By backing this out we can fix the "don't crash when called N times" case for the roll and this code will join the others in the pile of `AdaptCallbackForRepeating` uses to update.

Reply:
Rolled back in https://github.com/electron/electron/pull/29608/commits/ba70d10ff06d19109b9f3e5e1cb609e6861b62ac